pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -382,6 +387,26 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod) {
 		// update tr completed time
 		taskRun.Status.CompletionTime = &metav1.Time{Time: time.Now()}
 	}
+
+	updateTaskRunResourceResult(taskRun, pod, resourceLister, kubeclient, logger)
+}
+
+func updateTaskRunResourceResult(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLister listers.PipelineResourceLister, kubeclient kubernetes.Interface, logger *zap.SugaredLogger) {
+	if resources.TaskRunHasOutputImageResource(resourceLister.PipelineResources(taskRun.Namespace).Get, taskRun) && taskRun.IsSuccessful() {
+		for _, container := range pod.Spec.Containers {
+			if strings.HasPrefix(container.Name, imageDigestExporterContainerName) {
+				req := kubeclient.CoreV1().Pods(taskRun.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{Container: container.Name})
+				logContent, err := req.Do().Raw()
+				if err != nil {
+					logger.Errorf("Error getting output from image-digest-exporter for %s/%s: %s", taskRun.Name, taskRun.Namespace, err)
+				}
+				err = resources.UpdateTaskRunStatusWithResourceResult(taskRun, logContent)
```

Comment:
I've been trying to find an alternative to using the log file to share data, so that we could keep the log file to log things. The best idea I could come up with so far is to use the `imagedigestreporter` to update the TaskRun via patch and store the image digest there. 
Did you consider any other alternative yet?

Reply:
There was a [doc](https://docs.google.com/document/d/12loyvCrzKePoEzGEmKOOBg83yiDjrWngiNMPFtIBzYs/edit?usp=sharing) sent out a while back:  and this PR was based on the feedback I got.

Reply:
i know what you mean @afrittoli ! it seems a bit odd but it also seems like the best option at the moment - but fortunately since this is an implementation detail, I feel like we're safe to go ahead with this and if we come across a better idea we can change it later :D